modules/cli/src/main/scala/scala/cli/commands/publish/PublishSetup.scala

```diff
@@ -0,0 +1,316 @@
+
+      val pureJava = sources.hasJava && !sources.hasScala
+
+      (pureJava, sources.buildOptions.notForBloopOptions.publishOptions)
+    }
+
+    val backend = ScalaCliSttpBackend.httpURLConnection(logger)
+
+    val checksInputOpt = options.checks.map(_.trim).filter(_.nonEmpty).filter(_ != "all")
+    val checkKinds = checksInputOpt match {
+      case None => OptionCheck.Kind.all.toSet
+      case Some(checksInput) =>
+        OptionCheck.Kind.parseList(checksInput)
+          .left.map { unrecognized =>
+            System.err.println(s"Unrecognized check(s): ${unrecognized.mkString(", ")}")
+            sys.exit(1)
+          }
+          .merge
+          .toSet
+    }
+
+    val missingFields =
+      OptionChecks.checks(options, configDb, inputs.workspace, coursierCache, logger, backend)
+        .filter(check => checkKinds(check.kind))
+        .flatMap {
+          check =>
+            if (check.check(publishOptions)) {
+              logger.debug(s"Found field ${check.fieldName}")
+              Nil
+            }
+            else {
+              logger.debug(s"Missing field ${check.fieldName}")
+              Seq(check)
+            }
+        }
+
+    if (missingFields.nonEmpty) {
+      val count = missingFields.length
+      logger.message(s"$count ${if (count > 1) "options" else "option"} need to be set")
+      for (check <- missingFields)
+        logger.debug(s"  ${check.fieldName}")
+      logger.message("")
+    }
+
+    lazy val (ghRepoOrg, ghRepoName) = GitRepo.ghRepoOrgName(inputs.workspace, logger)
+      .orExit(logger)
+
+    lazy val token = options.token
+      .orElse(configDb.get(Keys.ghToken).orExit(logger))
+      .map(_.get())
+      .getOrElse {
+        System.err.println(
+          s"No GitHub token passed, please specify one via --token env:ENV_VAR_NAME or --token file:/path/to/token, " +
+            s"or by setting ${Keys.ghToken.fullName} in the config."
+        )
+        sys.exit(1)
+      }
+
+    if (options.check)
+      if (missingFields.isEmpty)
+        logger.message("Setup fine for publishing")
+      else {
+        logger.message("Found missing config for publishing")
+        sys.exit(1)
+      }
+    else {
+
+      val missingFieldsWithDefaults = missingFields
+        .map { check =>
+          check.defaultValue().map((check, _))
+        }
+        .sequence
+        .left.map(CompositeBuildException(_))
+        .orExit(logger)
+
+      lazy val secretNames = {
+        val secretList = SecretList.list(
+          ghRepoOrg,
+          ghRepoName,
+          token,
+          backend,
+          logger
+        ).orExit(logger)
+
+        secretList.secrets.map(_.name).toSet
+      }
+
+      val missingSetSecrets = missingFieldsWithDefaults
+        .flatMap {
+          case (_, default) => default.ghSecrets
+        }
+        .filter(s => s.force || !secretNames.contains(s.name))
+
+      if (missingSetSecrets.nonEmpty) {
+
+        logger.message("")
```

Comment:
?

Reply:
That's to print an empty line, for readability. I just pushed some comments for that.